cli/cmd/inject_test.go

```diff
@@ -201,25 +192,17 @@ func testInjectCmd(t *testing.T, tc injectCmd) {
 	if exitCode != tc.exitCode {
 		t.Fatalf("Expected exit code to be %d but got: %d", tc.exitCode, exitCode)
 	}
-
-	actualStdOutResult := outBuffer.String()
-	expectedStdOutResult := readOptionalTestFile(t, tc.stdOutGoldenFileName)
-	if expectedStdOutResult != actualStdOutResult {
-		t.Errorf("Result mismatch.\nExpected: %s\nActual: %s", expectedStdOutResult, actualStdOutResult)
+	if tc.stdOutGoldenFileName != "" {
+		testDiff(t, tc.stdOutGoldenFileName, outBuffer.String())
 	}
```

Comment:
For parity with the previous implementation, this should assert the output buffer is empty if no golden output file is given. Something like:

```diff
diff --git a/cli/cmd/inject_test.go b/cli/cmd/inject_test.go
index def9298b..76126848 100644
--- a/cli/cmd/inject_test.go
+++ b/cli/cmd/inject_test.go
@@ -194,6 +194,8 @@ func testInjectCmd(t *testing.T, tc injectCmd) {
 	}
 	if tc.stdOutGoldenFileName != "" {
 		testDiff(t, tc.stdOutGoldenFileName, outBuffer.String())
+	} else if outBuffer.Len() != 0 {
+		t.Fatalf("Expected no standard output, but got: %s", outBuffer)
 	}
 
 	stdErrGoldenFileName := tc.stdErrGoldenFileName
```